qiskit/transpiler/target.py

```diff
@@ -1176,6 +1184,212 @@ def __str__(self):
+            qubit_properties=qubit_properties,
+        )
+        name_mapping = get_standard_gate_name_mapping()
+        if custom_name_mapping is not None:
+            name_mapping.update(custom_name_mapping)
+
+        if coupling_map is None:
+            for gate in basis_gates:
+                if gate not in name_mapping:
+                    raise KeyError(
+                        f"The specified basis gate: {gate} is not present in the standard gate "
+                        "names or a provided custom_name_mapping"
+                    )
+                target.add_instruction(name_mapping[gate], name=gate)
+        else:
+            one_qubit_gates = []
+            two_qubit_gates = []
+            global_ideal_variable_width_gates = []  # pylint: disable=invalid-name
+            if num_qubits is None:
+                num_qubits = len(coupling_map.graph)
+            for gate in basis_gates:
+                if gate not in name_mapping:
+                    raise KeyError(
+                        f"The specified basis gate: {gate} is not present in the standard gate "
+                        "names or a provided custom_name_mapping"
+                    )
+                gate_obj = name_mapping[gate]
+                if gate_obj.num_qubits == 1:
+                    one_qubit_gates.append(gate)
+                elif gate_obj.num_qubits == 2:
+                    two_qubit_gates.append(gate)
+                elif inspect.isclass(gate_obj):
+                    global_ideal_variable_width_gates.append(gate)
+                else:
+                    raise TranspilerError(
+                        f"The specified basis gate: {gate} has an invalid number of qubits: "
+                        f"{gate_obj.num_qubits}"
+                    )
+            for gate in one_qubit_gates:
+                gate_properties = {}
+                for qubit in range(num_qubits):
+                    error = None
+                    duration = None
+                    calibration = None
+                    if instruction_durations is not None:
+                        try:
+                            duration = instruction_durations.get(gate, qubit, unit="s")
+                        except TranspilerError:
+                            duration = None
+                    if backend_properties is not None:
+                        if duration is None:
+                            try:
+                                duration = backend_properties.gate_length(gate, qubit)
+                            except BackendPropertyError:
+                                duration = None
+                        try:
+                            error = backend_properties.gate_error(gate, qubit)
+                        except BackendPropertyError:
+                            error = None
+                    if inst_map is not None:
+                        try:
+                            calibration = inst_map.get(gate, qubit)
```

Comment:
```suggestion
                            calibration = inst_map. _get_calibration_entry(gate, qubit)
```
This protected method allows bypassing generation of Schedule.
https://github.com/Qiskit/qiskit-terra/blob/5672c70649b4f83a6c7146c68a6a543d4a3908fd/qiskit/pulse/instruction_schedule_map.py#L197-L201

but here is a difficult question. When a user provides inst_map, it can override backend ISA and it has the highest priority for duration. However, duration cannot be obtained without parsing the calibration entry (i.e. converting Qobj into Schedule). What you may want to do is something like
https://github.com/Qiskit/qiskit-terra/blob/5672c70649b4f83a6c7146c68a6a543d4a3908fd/qiskit/transpiler/target.py#L492-L498
here you can check the `user_provided` attribute (ignore the second condition), and call `.get_schedule` only when the entry is user provided. Usually user provided schedule is already `Schedule` or `ScheduleBlock` format and there is very small overhead to get duration value. Backend provided schedules are not attached as calibration, so you can safely ignore them.

One edge case here is when user provided `ScheduleBlock` has variable duration (such as pulse efficient Rzx gate) and duration can be obtained after gate parameter assignment. So you may need to implement [try-except](https://github.com/Qiskit/qiskit-terra/blob/5672c70649b4f83a6c7146c68a6a543d4a3908fd/qiskit/pulse/transforms/canonicalization.py#L45-L48) block to make it more robust.